### DISCUSSION TITLE
CSP tick processing performance evaluation `NON_COLLAPSING `

### DISCUSSION THREAD

**Discussion by kootsZhin** [Point72/csp#349] [Q&A] [answered]
Thanks for making `csp` and it has been really interesting playing with it so far!
 
Currently I am trying load my csp with high frequency data under `NON_COLLAPSING` mode, assuming my average engine cycle time >>> tick frequency, I would assume there an existence of "unticked stack of data" - is there a way for us to access the size of the stack in run time?

**Comment by robambalu** [comment, marked as answer]
There’s no way to access the current queue size in realtime at the moment, but if your data is timestamped you can compare the data time to csp.now() to get a sense of how far behind the engine is

**Reply by kootsZhin** [reply]
Got it, thank you so much!
